docs/csharp/whats-new/csharp-8.md

```diff
@@ -0,0 +1,314 @@
+
+    public void Deconstruct(out int x, out int y) =>
+        (x, y) = (X, Y);
+}
+```
+
+The following method uses the **positional pattern** to extract the values of `x` and `y`. Then, it uses a `when` clause to determine the quadrant of the point:
+
+```csharp
+static string Quadrant(Point p) => p switch
+{
+    (0, 0) => "origin",
+    (var x, var y) when x > 0 && y > 0 => "Quadrant 1",
+    (var x, var y) when x < 0 && y > 0 => "Quadrant 2",
+    (var x, var y) when x < 0 && y < 0 => "Quadrant 3",
+    (var x, var y) when x > 0 && y < 0 => "Quadrant 4",
+    (var x, var y) => "on a border",
+    _ => "unknown"
+};
+```
+
+## using declarations
+
+A **using declaration** is a variable declaration preceded by the `using` keyword. It tells the compiler that the variable being declared should be disposed at the end of the enclosing scope. For example, consider the following code that writes a text file:
+
+```csharp
+static void WriteLinesToFile(IEnumerable<string> lines)
+{
+    using (var file = new System.IO.StreamWriter("WriteLines2.txt"))
+    {
+        foreach (string line in lines)
+        {
+            // If the line doesn't contain the word 'Second', write the line to the file.
+            if (!line.Contains("Second"))
+            {
+                file.WriteLine(line);
+            }
+        }
+    } // file is disposed here
+}
+```
+
+A few braces are removed by replacing the `using` statement with a `using` declaration:
+
+```csharp
+static void WriteLinesToFile(IEnumerable<string> lines)
+{
+    using var file = new System.IO.StreamWriter("WriteLines2.txt");
+    foreach (string line in lines)
+    {
+        // If the line doesn't contain the word 'Second', write the line to the file.
+        if (!line.Contains("Second"))
+        {
+            file.WriteLine(line);
+        }
+    }
+// file is disposed here
+}
+```
+
+Using declarations provide the same behavior as [using statements](../language-reference/keywords/using-statement.md) when exceptions are thrown: the `Dispose` call is in a `finally` block.
+
+## Static local functions
+
+You can now add the `static` modifier to local functions to ensure that local function doesn't capture (reference) any variables from the enclosing scope. Doing so generates `CS8421`, "A static local function can't contain a reference to <variable>." 
+
+Consider the following code. The local function `LocalFunction` accesses the variable `y`, declared in the enclosing scope (the method `M`). Therefore, `LocalFunction` can't be declared with the `static` modifier:
+
+```csharp
+int M()
+{
+    int y;
+    LocalFunction();
+    return y;
+
+    void LocalFunction() => y = 0;
+}
+```
+
+## Disposable ref structs
+
+A `struct` declared with the `ref` modifier may not implement any interfaces and so cannot implement <xref:System.IDisposable>. Therefore, to enable a `ref struct` to be disposed, it must have an accessible `void Dispose()` method.
```

Comment:
I would clarify how `void Dispose()` gets called. Is this something that has to be done by hand? Will the runtime do it for you? How does it interact with using statements? In C# 7 you get an error if you try to use an object that doesn't implement IDisposable: `using (var a = new ObjNotImplementIDisposable)`: did that change with C# 8?